2010-01-05  Samuel Thibault  <samuel.thibault@ens-lyon.org>

	* sysdeps/mach/hurd/bits/posix_opt.h
	(_POSIX_PRIORITY_SCHEDULING): Define macro to -1.
	(_POSIX_CHOWN_RESTRICTED): Define macro to 0.
	(_POSIX_NO_TRUNC): Define macro to 0.
	(_POSIX_SYNCHRONIZED_IO): Define macro to 0.
	(_XOPEN_REALTIME): Define macro to -1.
	(_XOPEN_REALTIME_THREADS): Define macro to -1.
	(_XOPEN_SHM): Undefine macro.
	[__USE_XOPEN2K8] (_POSIX_THREAD_ROBUST_PRIO_INHERIT): Define
	macro to -1.
	[__USE_XOPEN2K8] (_POSIX_THREAD_ROBUST_PRIO_PROTECT): Define
	macro to -1.
	(_POSIX_ASYNC_IO): Define macro to 1.
	(_POSIX_PRIORITIZED_IO): Define macro to -1.
	(_POSIX_SPIN_LOCKS): Define macro to -1.

---
 bits/posix_opt.h |   36 ++++++++++++++++++++++++++----------
 spawni.c         |    2 +-
 2 files changed, 27 insertions(+), 11 deletions(-)

diff --git a/sysdeps/mach/hurd/bits/posix_opt.h b/sysdeps/mach/hurd/bits/posix_opt.h
index 775c921..03ce39e 100644
--- a/sysdeps/mach/hurd/bits/posix_opt.h
+++ b/sysdeps/mach/hurd/bits/posix_opt.h
@@ -31,8 +31,9 @@
 /* Processes have a saved set-user-ID and a saved set-group-ID.  */
 #define	_POSIX_SAVED_IDS	1
 
-/* Synchronizing file data is supported, but msync is missing.  */
-#undef _POSIX_SYNCHRONIZED_IO
+/* Priority scheduling is not supported.  */
+#define	_POSIX_PRIORITY_SCHEDULING	-1
+
 
 /* The fsync function is present.  */
 #define	_POSIX_FSYNC	200809L
@@ -56,10 +57,18 @@
 
 /* Different Hurd filesystems might do these differently.
    You must query the particular file with `pathconf' or `fpathconf'.  */
-#undef _POSIX_CHOWN_RESTRICTED	/* Only root can change owner of file?  */
-#undef _POSIX_NO_TRUNC		/* Overlong file names get error?  */
-#undef _POSIX_SYNC_IO		/* File supports O_SYNC et al?  */
+#define _POSIX_CHOWN_RESTRICTED		0 /* Only root can change owner of file?  */
+#define _POSIX_NO_TRUNC			0 /* Overlong file names get error?  */
+#define _POSIX_SYNCHRONIZED_IO		0 /* File supports O_SYNC et al?  */
+
+/* X/Open realtime support is not available.  */
+#define _XOPEN_REALTIME	-1
+
+/* X/Open thread realtime support is available.  */
+#define _XOPEN_REALTIME_THREADS	-1
 
+/* XPG4.2 shared memory is not supported.  */
+#undef	_XOPEN_SHM
 
 /* We do not have the POSIX threads interface.  */
 #define _POSIX_THREADS	-1
@@ -72,6 +81,12 @@
 #define _POSIX_THREAD_PRIORITY_SCHEDULING	-1
 #define _POSIX_THREAD_ATTR_STACKSIZE		-1
 #define _POSIX_THREAD_ATTR_STACKADDR		-1
+#define _POSIX_THREAD_PRIO_INHERIT	-1
+#define _POSIX_THREAD_PRIO_PROTECT	-1
+#ifdef __USE_XOPEN2K8
+# define _POSIX_THREAD_ROBUST_PRIO_INHERIT	-1
+# define _POSIX_THREAD_ROBUST_PRIO_PROTECT	-1
+#endif
 #define _POSIX_SEMAPHORES			-1
 
 /* Real-time signals are not yet supported.  */
@@ -79,8 +94,11 @@
 
 /* Asynchronous I/O might supported with the existing ABI.  */
 #define _POSIX_ASYNCHRONOUS_IO	0
+#define _POSIX_ASYNC_IO		1
 /* Alternative name for Unix98.  */
 #define _LFS_ASYNCHRONOUS_IO	_POSIX_ASYNCHRONOUS_IO
+/* Support for prioritization is not available.  */
+#define _POSIX_PRIORITIZED_IO	-1
 
 /* The LFS support in asynchronous I/O is also available.  */
 #define _LFS64_ASYNCHRONOUS_IO	_POSIX_ASYNCHRONOUS_IO
@@ -111,6 +129,9 @@
 /* We cannot support the Timeouts option without _POSIX_THREADS.  */
 #define _POSIX_TIMEOUTS	-1
 
+/* We do not support spinlocks.  */
+#define _POSIX_SPIN_LOCKS	-1
+
 /* The `spawn' function family is supported.  */
 #define _POSIX_SPAWN	200809L
 
@@ -157,9 +178,4 @@
 /* Typed memory objects are not available.  */
 #define _POSIX_TYPED_MEMORY_OBJECTS	-1
 
-/* No support for priority inheritance or protection so far.  */
-#define _POSIX_THREAD_PRIO_INHERIT	-1
-#define _POSIX_THREAD_PRIO_PROTECT	-1
-
-
 #endif /* bits/posix_opt.h */
diff --git a/sysdeps/mach/hurd/spawni.c b/sysdeps/mach/hurd/spawni.c
index 244ca2d..150b2cf 100644
--- a/sysdeps/mach/hurd/spawni.c
+++ b/sysdeps/mach/hurd/spawni.c
@@ -264,7 +264,7 @@ __spawni (pid_t *pid, const char *file,
   if ((flags & POSIX_SPAWN_SETSIGMASK) != 0)
     ints[INIT_SIGMASK] = attrp->__ss;
 
-#ifdef _POSIX_PRIORITY_SCHEDULING
+#if _POSIX_PRIORITY_SCHEDULING > 0
   /* Set the scheduling algorithm and parameters.  */
 # error implement me
   if ((flags & (POSIX_SPAWN_SETSCHEDPARAM | POSIX_SPAWN_SETSCHEDULER))
